deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py

```diff
@@ -891,23 +911,29 @@ class Sender:
         >>> import wazuh_testing.tools.agent_simulator as ag
         >>> manager_address = "172.17.0.2"
         >>> agent = ag.Agent(manager_address, "aes", os="debian8", version="4.2.0")
-        >>> sender = ag.Sender(manager_address, protocol="tcp")
+        >>> sender = ag.Sender(manager_address, protocol=TCP)
     """
-    def __init__(self, manager_address, manager_port='1514', protocol='TCP'):
+    def __init__(self, manager_address, manager_port='1514', protocol=TCP):
         self.manager_address = manager_address
         self.manager_port = manager_port
         self.protocol = protocol.upper()
-        if self.protocol == 'TCP':
+        if is_tcp(self.protocol):
             self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             self.socket.connect((self.manager_address, int(self.manager_port)))
-        if self.protocol == 'UDP':
+        if is_udp(self.protocol):
             self.socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
 
     def send_event(self, event):
-        if self.protocol == 'TCP':
+        if is_tcp(self.protocol):
             length = pack('<I', len(event))
-            self.socket.send(length + event)
-        if self.protocol == 'UDP':
+            try:
+                self.socket.send(length + event)
+            except BrokenPipeError:
+                logging.warning(f"Broken Pipe error while sending event. Creating new socket...")
+                sleep(5)
+                self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                self.socket.connect((self.manager_address, int(self.manager_port)))
```

Comment:
I think that the `send` is missing here

Reply:
Yes, you're right, thanks! https://github.com/wazuh/wazuh-qa/pull/1106/commits/c6c8bd3ac9e94d39e6045c73a0a8ca6df73a9080